### PR TITLE
Improve tests

### PR DESCRIPTION
**Pull request by bedeho**
- [x] Move tests and mocking out of single lib file.
- [x] Properly mock membership module as a full rust module, and drop the static unsafe hashmap approach, as suggested in #11 .
- [x] Add a test using the new membership module mock

### CODE REVIEW

**Comment by mnaamani** [comment]
after merged PRs #21 and #22, this needs a fix:

```
error[E0308]: mismatched types
   --> src/tests.rs:139:31
    |
139 |             title: vec![b'X'; genesis_config.category_title_constraint.max() + 1],
    |                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected usize, found u16
```

File: src/mock.rs [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+    type Moment = u64;
+    type OnTimestampSet = ();
+}
+
+impl Trait for Runtime {
+    type Event = ();
+    type MembershipRegistry = registry::TestMembershipRegistryModule;
+}
+
+// Here are a few testing utilities and fixtures, will reorganize
```
Comment (mnaamani) [inline src/mock.rs]: These fixtures are not used inside mock.rs, as they have been moved to tests.rs

**Review by mnaamani — CHANGES_REQUESTED** [review]
Unused fixtures can be removed from mock.rs.
Apart from that tests are all passing.

File: src/mock.rs (hunk 1 above)
Comment (bedeho) [reply src/mock.rs]: Oh, thanks for catching this!

**Review by mnaamani — APPROVED** [review]

**Pull request merged**